lib/IDE/CodeCompletion.cpp

```diff
@@ -6708,7 +6710,130 @@ static void deliverCompletionResults(CodeCompletionContext &CompletionContext,
   Consumer.handleResultsAndModules(CompletionContext, RequestedModules, DC);
 }
 
-void deliverUnresolvedMemberResults(
+/// Populates a vector of parameters to suggest along with a vector of types to
+/// match the lookup results against.
+///
+/// \Returns true if global lookup should be performed.
+static bool addPossibleParams(
+    const ArgumentTypeCheckCompletionCallback::Result &Ret,
+    SmallVectorImpl<PossibleParamInfo> &Params, SmallVectorImpl<Type> &Types) {
+  ArrayRef<AnyFunctionType::Param> ParamsToPass =
+      Ret.FuncTy->getAs<AnyFunctionType>()->getParams();
+  ParameterList *PL = nullptr;
+
+  if (auto *FD = dyn_cast<AbstractFunctionDecl>(Ret.FuncD)) {
+    PL = FD->getParameters();
+  } else if (auto *SD = dyn_cast<SubscriptDecl>(Ret.FuncD)) {
+    PL = SD->getIndices();
+  } else {
+    assert(false && "Unhandled decl type?");
+    return true;
+  }
+  assert(PL->size() == ParamsToPass.size());
+
+  if (!Ret.ParamIdx)
+    return true;
+
+  if (Ret.HasLabel) {
+    Types.push_back(Ret.ExpectedType);
+    return true;
+  }
```

Comment:
You can move those two cases before retrieving `PL` to avoid some work in case there is a label.